### DISCUSSION TITLE
🚁  Recommendations for running background jobs? 🧑‍🏭

### DISCUSSION THREAD

**Discussion by christopherpickering** [remix-run/remix#5078] [Q&A] [answered]
Hi! I'm looking for advice on the best approach to do background jobs. In python django projects I'm generally using celery.. is there a good node tool you would recommend?

- for sending confirmation emails
- for watching an email inbox
- running background database calculations with prisma
- for running scheduled background jobs, such as loading user profiles from external source, etc
- wanna be able to monitor job status from remix

Any tips or suggestions or example links are appreciated. loving remix. 🤩

**Comment by kiliman** [comment, marked as answer]
I'm currently using Quirrel https://quirrel.dev. They no longer provide a managed service, but it's really easy to self-host.

I'm hosting it on Fly via Docker and their Redis Upstash service. Everything works fine even on Fly's free tier.

**Reply by christopherpickering** [reply]
Nice, thanks for the suggestion, I will look into it.

**Reply by moishinetzer** [reply]
Hey @kiliman is there any chance you could make a public repo for a starter template for this? That would be super awesome 😄 (love all your other work as well btw)

**Reply by kiliman** [reply]
Thanks. I'll try and get something posted later this week.

**Comment by chronark** [comment]
Hey, check out https://upstash.com/qstash
disclaimer: I work at upstash and built qstash

I've also written a blog about one of your required use cases: https://upstash.com/blog/qstash-periodic-data-updates

happy to answer any questions you have

**Reply by christopherpickering** [reply]
thanks!

**Reply by nilooy** [reply]
qstash seems super nice but one issue that bothers me is using ngrok for local development unlike quirrel which seems to be a extra dependency.

**Comment by christopherpickering** [comment]
I ended up going with quirrel, its great and I'd recommend it.
